public/app/plugins/panel/geomap/layers/data/routeLayer.tsx

```diff
@@ -87,12 +92,50 @@ export const routeLayer: MapLayerRegistryItem<RouteConfig> = {
       vectorLayer.setStyle(routeStyle(style.base));
     } else {
       vectorLayer.setStyle((feature: FeatureLike) => {
+        // TODO better understand how we want to apply color logic functions
         const idx = feature.get('rowIndex') as number;
         const dims = style.dims;
         if (!dims || !isNumber(idx)) {
           return routeStyle(style.base);
         }
 
+        const styles = [];
+        const geom = feature.getGeometry();
+
+        if (geom instanceof SimpleGeometry && dims.color) {
```

Comment:
I think you can also check it the config color is fixed -- since I *think* dims.color would exist all the time